test_conformance/subgroups/subgroup_common_templates.h

```diff
@@ -565,6 +589,15 @@ template <typename Ty, ShuffleOp operation> struct SHF
                             if (l >= ns) skip = true;
                             tr_idx = i + l;
                             break;
+                        // rotate - treat l as delta
+                        case ShuffleOp::rotate:
+                            tr_idx = (i + l) % test_params.subgroup_size;
+                            break;
+                        case ShuffleOp::clustered_rotate: {
+                            tr_idx = ((i & ~(test_params.cluster_size - 1))
+                                      + ((i + l) % test_params.cluster_size));
+                            break;
```

Comment:
Thanks, @jlewis-austin, for your review under PR #1435.  I've tidied up the code as you suggested.  It was previously:

```
                        // rotate - treat l as delta
                        case ShuffleOp::rotate:
                        case ShuffleOp::clustered_rotate: {
                            if (operation == ShuffleOp::rotate)
                            {
                                tr_idx = (i + l) % test_params.subgroup_size;
                            }
                            else // operation == ShuffleOp::clustered_rotate
                            {
                                tr_idx =
                                    ((i & ~(test_params.cluster_size - 1))
                                     + ((i + l) % test_params.cluster_size));
                            }
                            break;
                        }
```
As you pointed out:
> these two cases don't share any code; it seems cleaner to avoid the fall through and just handle each in its own block